proto/substrait/algebra.proto

```diff
@@ -726,6 +730,47 @@ message NestedLoopJoinRel {
   substrait.extensions.AdvancedExtension advanced_extension = 10;
 }
 
+message DelimGetRel{
+   RelCommon common = 1;
+   repeated  Type  delim_types = 2;
```

Comment:
If the inputs are identified. then the types could be related to the input fields.  Then here would be a series of field references to say which fields are being used.  Ideally the logic for this relation could be determined without knowledge of the DelimJoinRelation.